### PR TITLE
fix: use tooltip on money request header avatar

### PR DESCRIPTION
**Pull request by allroundexperts**
### Details

This PR adds tooltips to the payee and the payer in the `MoneyRequestHeader` component.

### Fixed Issues

$ https://github.com/Expensify/App/issues/19341
PROPOSAL: https://github.com/Expensify/App/issues/19341#issuecomment-1555922209

### Tests

1. Open the App and by clicking on the `+` icon in the LHN, send money to any user.
2. In the chat that opens with the user after sending money, click on the IOU preview message.
3. This opens the IOU transaction report page. 

Verify that the tooltip shows in the header by moving the cursor over the sender and receiver avatars + names.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

1. Open the App and by clicking on the `+` icon in the LHN, send money to any user.
2. In the chat that opens with the user after sending money, click on the IOU preview message.
3. This opens the IOU transaction report page. 

Verify that the tooltip shows in the header by moving the cursor over the sender and receiver avatars + names.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/30054992/0829331f-5785-4aeb-a8fd-c76032474599

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-06-01 at 3 27 02 PM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-06-01 at 3 26 42 PM

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/30054992/36329ccd-d1f3-47b5-9766-a144d322795f

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-06-01 at 3 36 09 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-06-01 at 3 36 45 PM

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
@allroundexperts We are using email not displayname in chat header. Shouldn't we do the same here in our case as well?

**Comment by sobitneupane** [comment]
@allroundexperts Looks like we don't show any tooltip when hovered over Workspace name in chat header. So, should we do the same for workspace in Money Request Header as well?

I think one way will be to show email for individual and nothing for Workspace. What is your opinion on this @youssef-lr?

**Comment by youssef-lr** [comment]
Yeah I don't think we need to show a tooltip for the workspace, as it'll basically be the same as the display name.

**Comment by sobitneupane** [comment]
@allroundexperts There are still multiple issues.

- First we decided not to show anything if its workspace (Please mimic chat header of workspace)

https://github.com/Expensify/App/assets/25876548/b9d823b1-4eb3-42c1-a093-47ec696a48c0



- Second please do consider subscript avatar. For avatar with subscript avatar (Workspace) we are showing two different tooltips. I don't think we would want to show any tooltip in subscript avatar.

https://github.com/Expensify/App/assets/25876548/13dfc9ef-f95c-4e13-a06e-50eabe03a33f


- Third requestor avatar should also show email
Screenshot 2023-06-06 at 17 09 25

File: src/components/AvatarWithDisplayName.js [hunk 1]
```diff
@@ -69,15 +70,20 @@ const AvatarWithDisplayName = (props) => {
                             mainTooltip={props.report.ownerEmail}
                             secondaryTooltip={subtitle}
                             size={props.size}
+                            noMargin
```
Comment (sobitneupane) [inline src/components/AvatarWithDisplayName.js:73]: This change doesn't look related to this PR.

File: src/components/AvatarWithDisplayName.js [hunk 2]
```diff
@@ -86,7 +92,7 @@ const AvatarWithDisplayName = (props) => {
                             tooltipEnabled
                             numberOfLines={1}
                             textStyles={[props.isAnonymous ? styles.headerAnonymousFooter : styles.headerText, styles.pre]}
-                            shouldUseFullTitle={isExpenseReport || props.isAnonymous}
+                            shouldUseFullTitle={props.isAnonymous}
```
[the remark this answers is not in the document]
Comment (youssef-lr) [reply src/components/AvatarWithDisplayName.js:95]: I think it's intentional to only display the full name based on those two conditions so I'd say let's remove that change from this PR.

File: src/components/AvatarWithDisplayName.js (hunk 1 above)
Comment (sobitneupane) [reply src/components/AvatarWithDisplayName.js:73]: I don't think we want this change either. If you think it's a bug than it's better to report in slack first. Let's try to keep this PR limited to what actually we are trying to handle.

**Review by sobitneupane — COMMENTED** [review]
I think you should rerun the tests and update the screenshots as well.

I found two issues:

- Tooltip not showing when money is requested to workspace
- We need to update tooltips to new format. Let's try to keep everything consistent with main/staging.

https://github.com/Expensify/App/assets/25876548/a4f4b26f-40ef-40e7-9673-f807a3147f57

**Comment by sobitneupane** [comment]
> Yeah I don't think we need to show a tooltip for the workspace, as it'll basically be the same as the display name.

Screenshot 2023-06-19 at 16 33 37

@allroundexperts I think we decided not to show tooltip for workspace name as it just shows the display name itself.

**Review by sobitneupane — COMMENTED** [review]
Tooltip in icon of concierge in chat header is showing default image. I tested it in main and staging. Main and staging are working as expected.
Screenshot 2023-06-20 at 13 29 20

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/c4bd27ee-d003-48a8-bb11-72ccb50c9eca

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-06-21 at 14 36 13

</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-06-21 at 14 37 22

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/8b8e824d-d4e7-418f-904f-79b179789f0e

</details>

<details>
<summary>iOS</summary>
Screenshot 2023-06-21 at 14 36 48

</details>

<details>
<summary>Android</summary>
Screenshot 2023-06-21 at 14 38 48

</details>

File: src/components/AvatarWithDisplayName.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sobitneupane) [reply src/components/AvatarWithDisplayName.js:94]: @allroundexperts This change has made native and web inconsistent.

[the remark this answers is not in the document]
Comment (sobitneupane) [reply src/components/AvatarWithDisplayName.js:94]: Screenshot 2023-06-21 at 18 14 01
Screenshot 2023-06-21 at 18 14 32

[the remark this answers is not in the document]
Comment (sobitneupane) [reply src/components/AvatarWithDisplayName.js:94]: bump @allroundexperts

[the remark this answers is not in the document]
Comment (sobitneupane) [reply src/components/AvatarWithDisplayName.js:94]: > I think it's intentional to only display the full name based on those two conditions so I'd say let's remove that change from this PR.

@allroundexperts As suggested by @youssef-lr I think the better idea is to remove this change from this PR. I cannot get how this change is related to the issue we are trying to solve.

**Comment by sobitneupane** [comment]
@allroundexperts Looks like we will no longer need these changes. We are going to update money request header in [this issue](https://github.com/Expensify/App/issues/20486).
